sycl/include/CL/sycl/pipes.hpp

```diff
@@ -12,11 +12,33 @@
 #include <CL/__spirv/spirv_types.hpp>
 #include <CL/sycl/stl.hpp>
 
+#ifndef __SYCL_DEVICE_ONLY__
+#include <iostream>
+#endif // __SYCL_DEVICE_ONLY__
+
 namespace cl {
 namespace sycl {
 
-template <class name, class dataT, int32_t min_capacity = 0> class pipe {
+// Utility to dump typeid
+template <typename T> class wrap {};
+template <typename Type>
+std::string __str_type_name() {
+  return std::string(typeid(wrap<Type>).name());
+}
+
+template <typename PipeName>
+inline void __pipe_printer(std::string Input, int32_t S, int32_t A, int32_t C) {
+  std::cout << Input << __str_type_name<PipeName>() << std::endl
```

Comment:
There should be an option to disable output to stdout. If I have a million reads and writes, I probably don't want to see all this output.

Reply:
I think it must be an option to control "debugging" and this features must be disabled by default.